src/sentry/preprod/size_analysis/tasks.py

```diff
@@ -0,0 +1,306 @@
+        if comparison.state in [
+            PreprodArtifactSizeComparison.State.PROCESSING,
+            PreprodArtifactSizeComparison.State.SUCCESS,
+            PreprodArtifactSizeComparison.State.FAILED,
+        ]:
+            logger.info(
+                "preprod.size_analysis.compare.existing_comparison",
+                extra={
+                    "head_artifact_size_metric_id": head_size_metric.id,
+                    "base_artifact_size_metric_id": base_size_metric.id,
+                },
+            )
+            return
+
+    except PreprodArtifactSizeComparison.DoesNotExist:
+        logger.info(
+            "preprod.size_analysis.compare.no_existing_comparison",
+            extra={
+                "head_artifact_size_metric_id": head_size_metric.id,
+                "base_artifact_size_metric_id": base_size_metric.id,
+            },
+        )
+        pass
+
+    try:
+        head_analysis_file = File.objects.get(id=head_size_metric.analysis_file_id)
+    except File.DoesNotExist:
+        logger.exception(
+            "preprod.size_analysis.compare.head_analysis_file_not_found",
+            extra={
+                "head_artifact_size_metric_id": head_size_metric.id,
+                "head_artifact_id": head_size_metric.preprod_artifact.id,
+            },
+        )
+        return
+
+    try:
+        base_analysis_file = File.objects.get(id=base_size_metric.analysis_file_id)
+    except File.DoesNotExist:
+        logger.exception(
+            "preprod.size_analysis.compare.base_analysis_file_not_found",
+            extra={
+                "base_artifact_size_metric_id": base_size_metric.id,
+                "base_artifact_id": base_size_metric.preprod_artifact.id,
+            },
+        )
+        return
+
+    head_size_analysis_results = SizeAnalysisResults.parse_raw(head_analysis_file.getfile().read())
+    base_size_analysis_results = SizeAnalysisResults.parse_raw(base_analysis_file.getfile().read())
+
+    logger.info(
+        "preprod.size_analysis.compare.start",
+        extra={
+            "head_artifact_size_metric_id": head_size_metric.id,
+            "base_artifact_size_metric_id": base_size_metric.id,
+        },
+    )
+
+    comparison = PreprodArtifactSizeComparison.objects.create(
+        head_size_analysis=head_size_metric,
+        base_size_analysis=base_size_metric,
+        organization_id=head_size_metric.preprod_artifact.project.organization.id,
+        state=PreprodArtifactSizeComparison.State.PROCESSING,
+    )
+
+    comparison_results = compare_size_analysis(
+        head_size_analysis=head_size_metric,
+        head_size_analysis_results=head_size_analysis_results,
+        base_size_analysis=base_size_metric,
+        base_size_analysis_results=base_size_analysis_results,
+    )
+
+    logger.info(
+        "preprod.size_analysis.compare.create_file",
+        extra={"comparison_id": comparison.id},
+    )
+    file = File.objects.create(
```

Comment:
This and the PreprodArtifactSizeComparison should be done in the same transaction.